File: Ui/angular10/src/Shared/task.ts
import { Member } from "./member";

export class Task {
    TaskId :number;
    MemberId :number;
    Description :string;
    DeliverDate:string;
    IsDone:boolean;
    Member?:any;
    constructor(_TaskId: number,_MemberId:number, _Description :string,_DeliverDate :string,_IsDone:boolean,_Member:any) {
        this.TaskId = _TaskId;
        this.MemberId = _MemberId;
        this.Description = _Description;
        this.IsDone = _IsDone;
        this.DeliverDate=_DeliverDate;
        this.Member=_Member;
}
}

File: DotNetTask/DotNetTask.Core/Models/Task.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace DotNetTask.Core.Models
{
   public class Task
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int TaskId { get; set; }
        [Required, ForeignKey("Member")]
        public int MemberId { get; set; }
        [Required, MaxLength(100), MinLength(10)]
        public string Description { get; set; }
        [Required]
        public string DeliverDate { get; set; }
        public bool IsDone { get; set; }
        public virtual Member Member { get; set; }
    }
}
